internal/interpreter/api.rs

```diff
@@ -464,16 +465,16 @@ pub(crate) fn normalize_identifier(ident: &str) -> Cow<'_, str> {
 /// assert_eq!(s.get_field("foo").cloned().unwrap().try_into(), Ok(45u32));
 /// ```
 #[derive(Clone, PartialEq, Debug, Default)]
-pub struct Struct(HashMap<String, Value>);
+pub struct Struct(HashMap<SmolStr, Value>);
 impl Struct {
     /// Get the value for a given struct field
     pub fn get_field(&self, name: &str) -> Option<&Value> {
         self.0.get(&*normalize_identifier(name))
     }
     /// Set the value of a given struct field
-    pub fn set_field(&mut self, name: String, value: Value) {
+    pub fn set_field(&mut self, name: SmolStr, value: Value) {
```

Comment:
the interpreter API is public API and can unfortunately not be changed. So it must continue using String in its public interface.  (in retrospect, this should have been a `&str` perhaps.)

This is valid for other changes in this file.

We can make an issue to change that with Slint 2.0